### ISSUE TITLE
NPE when trying to run 'aggregation'

### ISSUE THREAD

**Issue by iivalchev** [onyx-platform/onyx-examples#8]
When trying to run the `aggregation` example project in a REPL session by manually evaluating the expressions from the `aggregation.core` namespace I can see the following exception in the `onyx.log` after evaluating `(def submission (onyx.api/submit-job peer-config ...` on line 143

```
17-04-04 12:26:55 tado-laptop INFO [onyx.messaging.aeron.endpoint-status:56] - Started endpoint status on peer: 2d89d2f3-0ef6-115c-b456-1596fa5aa29d
17-04-04 12:26:55 tado-laptop INFO [onyx.messaging.aeron.endpoint-status:56] - Started endpoint status on peer: ddb69b30-e04a-6a9a-0ffd-3f7465d6cd3e
17-04-04 12:26:55 tado-laptop INFO [onyx.peer.coordinator:292] - Piggybacking coordinator on peer: ddb69b30-e04a-6a9a-0ffd-3f7465d6cd3e
17-04-04 12:26:55 tado-laptop INFO [onyx.messaging.aeron.endpoint-status:56] - Started endpoint status on peer: 38bc88c0-842d-02b4-12ed-91d081a52ea3
17-04-04 12:26:55 tado-laptop INFO [onyx.messaging.aeron.endpoint-status:56] - Started endpoint status on peer: [:coordinator #uuid "ddb69b30-e04a-6a9a-0ffd-3f7465d6cd3e"]
17-04-04 12:26:55 tado-laptop INFO [onyx.peer.task-lifecycle:346] - Job 6e50439d-e195-323c-d2c7-fc8706299304 {:job-id #uuid "6e50439d-e195-323c-d2c7-fc8706299304", :job-hash "ea9cbe4116113b7238dedb52ee30931f5e3651726d17128bb185dea89235f2e"} - Task {:id :in, :name :in, :ingress-tasks [], :egress-tasks #{:split-sentence}} - Peer 38bc88c0-842d-02b4-12ed-91d081a52ea3 - Recover pipeline checkpoint: nil
17-04-04 12:26:55 tado-laptop INFO [onyx.peer.task-lifecycle:381] - Job 6e50439d-e195-323c-d2c7-fc8706299304 {:job-id #uuid "6e50439d-e195-323c-d2c7-fc8706299304", :job-hash "ea9cbe4116113b7238dedb52ee30931f5e3651726d17128bb185dea89235f2e"} - Task {:id :out, :name :out, :ingress-tasks #{:count-words}, :egress-tasks #{}} - Peer 784073ba-dcc6-2cd5-b003-0bb89e1f3869 - Recover output pipeline checkpoint: nil
17-04-04 12:26:55 tado-laptop WARN [onyx.peer.task-lifecycle:131] -
                              java.lang.Thread.run              Thread.java:  745
java.util.concurrent.ThreadPoolExecutor$Worker.run  ThreadPoolExecutor.java:  617
 java.util.concurrent.ThreadPoolExecutor.runWorker  ThreadPoolExecutor.java: 1142
                                               ...                               
                 clojure.core.async/thread-call/fn                async.clj:  439
 onyx.peer.task-lifecycle/start-task-lifecycle!/fn       task_lifecycle.clj:  902
      onyx.peer.task-lifecycle/run-task-lifecycle!       task_lifecycle.clj:  448
                onyx.peer.task-lifecycle/iteration       task_lifecycle.clj:  429
    onyx.peer.task-lifecycle.TaskStateMachine/exec       task_lifecycle.clj:  826
            onyx.peer.task-lifecycle/recover-input       task_lifecycle.clj:  347
onyx.plugin.core-async.AbsCoreAsyncReader/recover!           core_async.clj:   21
                                clojure.core/deref                 core.clj: 2228
                         clojure.core/deref-future                 core.clj: 2208
java.lang.NullPointerException:
    clojure.lang.ExceptionInfo: Handling uncaught exception thrown inside task lifecycle :lifecycle/recover-input. Killing the job. -> Exception type: java.lang.NullPointerException. Exception message: null
       job-id: #uuid "6e50439d-e195-323c-d2c7-fc8706299304"
     metadata: {:job-id #uuid "6e50439d-e195-323c-d2c7-fc8706299304", :job-hash "ea9cbe4116113b7238dedb52ee30931f5e3651726d17128bb185dea89235f2e"}
      peer-id: #uuid "38bc88c0-842d-02b4-12ed-91d081a52ea3"
    task-name: :in

17-04-04 12:26:55 tado-laptop INFO [onyx.peer.task-lifecycle:452] - Job 6e50439d-e195-323c-d2c7-fc8706299304 {:job-id #uuid "6e50439d-e195-323c-d2c7-fc8706299304", :job-hash "ea9cbe4116113b7238dedb52ee30931f5e3651726d17128bb185dea89235f2e"} - Task {:id :out, :name :out, :ingress-tasks #{:count-words}, :egress-tasks #{}} - Peer 784073ba-dcc6-2cd5-b003-0bb89e1f3869 - Fell out of task lifecycle loop
17-04-04 12:26:55 tado-laptop INFO [onyx.peer.task-lifecycle:1038] - Job 6e50439d-e195-323c-d2c7-fc8706299304 {:job-id #uuid "6e50439d-e195-323c-d2c7-fc8706299304", :job-hash "ea9cbe4116113b7238dedb52ee30931f5e3651726d17128bb185dea89235f2e"} - Task {:id :out, :name :out, :ingress-tasks #{:count-words}, :egress-tasks #{}} - Peer 784073ba-dcc6-2cd5-b003-0bb89e1f3869 - Stopping task lifecycle.
```

**Comment by MichaelDrogalis** [comment]
@iivalchev Thanks! I'll get a look at this tonight.

**Comment by MichaelDrogalis** [comment]
@iivalchev Fixed in 909c327daf955a83ba1ab478c803b922dd2e976d and pushed to master/0.10.x. Thanks!

**Comment by iivalchev** [comment]
@MichaelDrogalis works now, thanks!
